src/Microsoft.DotNet.Build.Tasks.Packaging/src/GetApplicableAssetsFromPackageReports.cs

```diff
@@ -79,13 +85,13 @@ public override bool Execute()
 
             CompileAssets = compileAssets.ToArray();
             RuntimeAssets = runtimeAssets.ToArray();
+            BuildProjects = buildProjects.Distinct().Select(bp => bp.ToItem()).ToArray();
```

Comment:
This is all the task work that enabled me to build from the referencing project instead of from within the pkgproj, but I didn't include the targets change because it required some messy UndefineProperties.  You can see that here, I'm not submitting it at the moment: https://github.com/ericstj/buildtools/commit/9bd54529a8ef4b0c73d552aeae8598ff474996a9